### DISCUSSION TITLE
KhrSurface API not finding any native functions

### DISCUSSION THREAD

**Discussion by Fullbrik** [dotnet/Silk.NET#843] [Q&A]
# Summary
I'm following the tutorial over at [vulkan-tutorial.com](vulkan-tutorial.com), but in c# with Silk.Net.Vulkan and Silk.Net.GLFW. Most of the tutorial translated well, except for GetPhysicalDeviceSurfaceSupport which throws Silk.NET.Core.Loader.SymbolLoadingException: Native symbol not found (Symbol: vkGetPhysicalDeviceSurfaceSupportKHR). 

# Steps to reproduce
- Platform: Desktop (PopOS/Linux)
- Framework Version: .NET 6
- API: Vulkan
- API Version: Vulkan 1.2
1. Initialize vulkan instance with GLFW extensions
2. Enumerate over GetPhysicalDeviceQueueFamilyProperties
3. Check if a family index returns true when using GetPhysicalDeviceSurfaceSupport
4. Crash with the exception above

# Comments
I am new to vulkan, but decently know c#. I have tested the tutorial's c++ example, it compiled an ran, so the function does exist on my computer.

**Comment by HurricanKai** [comment]
How did you call the function? Did you specify the extension when creating the device?

**Comment by Fullbrik** [comment]
Thanks for getting back to me. I first call ` var glfwExtensions = _glfw.GetRequiredInstanceExtensions(out var glfwExtensionCount);` to get the extensions glfw needs, and pass that to my InstanceCreateInfo:

` var createInfo = new InstanceCreateInfo()
            {
                SType = StructureType.InstanceCreateInfo,
                PApplicationInfo = &appInfo,
                EnabledExtensionCount = extensionCount,
                PpEnabledExtensionNames = extensions,
                EnabledLayerCount = 0
            };`
(glfwExtensions/glfwExtensionCount and extensions/extensionCount are the same)

**Comment by Perksey** [comment]
Are you using the TryGetExtension method to get your KhrSurface?

**Comment by Fullbrik** [comment]
I call `if (!vk.TryGetInstanceExtension(_instance, out _surfaceApi)) throw new Exception("Failed to get surface api, which is required for managing vulkan windows.");`

**Comment by Perksey** [comment]
Everything seems to be in order then, I’m not sure why this is happening. Other than manually validating that GLFW is giving you the correct extensions (using SilkMarshal.PtrToStringArray and then writing that to Console or Debug) I’m not sure what to suggest

**Comment by Fullbrik** [comment]
I did that, and it returned VK_KHR_surface and VK_KHR_xcb_surface

**Comment by HurricanKai** [comment]
Can you run the SDK included vkcube? If not it's probably not our issue but a problem with your computer.
If yes, please send us a full repo (csproj + cs files) to run locally.

**Comment by Fullbrik** [comment]
I can run VKCube locally. 

I don't want to share the whole project, but I will share the stuff with Vulkan in it (Also pls ignore the stupid name). The error is in VulkanDeviceManager.cs, line 127. Link: [https://drive.google.com/file/d/1WsWFzBzquacGMImiwlyZkDXLDI-d-MYV/view?usp=sharing](https://drive.google.com/file/d/1WsWFzBzquacGMImiwlyZkDXLDI-d-MYV/view?usp=sharing)

**Comment by Perksey** [comment]
Going to migrate this to a discussion until we know for sure that this is a silk bug.

**Comment by KingoBoiii** [comment]
Did you remember to enable the device extension "VK_KHR_swapchain"?
